### PR TITLE
TECH-189: Create test configuration for "/{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/getOpenAPI" endpoint 

### PR DESCRIPTION
**Pull request by mgrochalskisoldevelo**
Created mocks and Cucumber tests for getOpenAPI endpoint.

**Description**
Create test configuration for /{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/getOpenAPI endpoint from https://github.com/GovStackWorkingGroup/bb-information-mediator/blob/main/api/govstack_im_service_metadata_api-0.3-swagger.json .This should include all the steps that are pointed out in the parent ticket.

TICKET: https://govstack-global.atlassian.net/browse/TECH-189

### CODE REVIEW

File: test/features/get_open_api.feature [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Feature: Retrieve openAPI description of the specified REST service
+
+  Retrieve openAPI description of the specified REST service
+
+  @smoke
+  Scenario: Retrieve the openAPI description of the specified REST service smoke type test
+
+    Given User wants to retrieve the openAPI description of the specified REST service
+    When User sends GET request with given "identifier" as GovStackInstance "identifier" as memberClass "alp4aNum3r1c" as memberCode "alp4aNum3r1c" as applicationCode
+    Then User receives a resonse
```
Comment (KarolinaKopacz) [inline test/features/get_open_api.feature]:
```suggestion
    Then User receives a response
```

File: test/features/get_open_api.feature [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    And The response should match json schema
+    And The response should have status 200
+    And The response should be returned in a timely manner
```
Comment (KarolinaKopacz) [inline test/features/get_open_api.feature]: I suggest to change the order of AND. First "...should be returned in a timely manner", then check the response status, then header content-type or similar and at the end json schema :)

File: test/features/get_open_api.feature [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    | a552example      | zzz321a     | m3mb3rc0d3   | aa15            |
+    | valid1dentifi3r  | rand0mid3nt | str1ng       | azxcv55         |
+
+
+  @unit @negativeregression
+  Scenario Outline: Unable to retrieve the openAPI description of the specified REST service
+
+    Given User wants to retrieve the openAPI description of the specified REST service
+    When User sends GET request with given "<GovStackInstance>" as GovStackInstance "<memberClass>" as memberClass "<memberCode>" as memberCode "<applicationCode>" as applicationCode
+    Then User receives a resonse
```
Comment (KarolinaKopacz) [inline test/features/get_open_api.feature]: typo

File: test/features/support/get_open_api.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+const baseUrl = localhost + getOpenApiUrl;
+
+Before(() => {
+  specGetOpenAPI = spec();
+});
+
+// Scenario: Retrieve the openAPI description of the specified REST service smoke type test
+Given(
+  'User wants to retrieve the openAPI description of the specified REST service',
+  () => ''
```
Comment (KarolinaKopacz) [inline test/features/support/get_open_api.js]: Why empty string?

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

**Review by KarolinaKopacz — APPROVED** [review]

**Review by aleksander-reitsakas — CHANGES_REQUESTED** [review]
getOpenApi has one query parameter serviceCode

**Review by aleksander-reitsakas — CHANGES_REQUESTED** [review]
Here also _.:- alowed

File: test/features/support/get_open_api.js [hunk 5]
```diff
@@ -53,16 +58,20 @@ Then('The response should match json schema', () =>
...
-// Scenario: Unable to retrieve the openAPI description of the specified REST service
+// Scenario Outline: Unable to retrieve the openAPI description of the specified REST service of an invalid path parameter
 Then('The response should have status 400', () =>
   specGetOpenAPI.response().to.have.status(400)
 );
 
-After(() => {
+// Scenario Outline: Unable to retrieve the openAPI description of the specified REST service of an invalid serviceCode parameter
+
+// Scenario: Unable to retrieve the openAPI description of the specified REST service because of missing serviceCode parameter
```
Comment (KarolinaKopacz) [inline test/features/support/get_open_api.js]: There is no code below. If the step what you use here is defined in line 67-69 then better to write this scenario title above it, somehow merge it or write comment below saying where the code is written

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

**Review by KarolinaKopacz — APPROVED** [review]

**Review by aleksander-reitsakas — APPROVED** [review]

**Pull request merged**